### PR TITLE
Clean up observables

### PR DESCRIPTION
**Pull request by blink1073**
- Removes `Vector` class
- Renames `Vector`-based classes to `List` and mimics the Google [CollaborativeList](https://developers.google.com/google-apps/realtime/reference/gapi.drive.realtime.CollaborativeList) API
- Unifies the handling of `IObservableFoo.IChangedArgs`

Re-hash of https://github.com/jupyterlab/jupyterlab/pull/1959 without affecting the signal payloads.

### CODE REVIEW

**Comment by ian-r-rose** [comment]
This looks good to me. Thanks for cleaning it up.

File: packages/notebook/src/celllist.ts [hunk 1]
```diff
@@ -182,8 +144,8 @@ class CellList implements IObservableUndoableVector<ICellModel> {
    * #### Undefined Behavior
    * An `index` which is non-integral or out of range.
    */
-  at(index: number): ICellModel {
-    return this._cellMap.get(this._cellOrder.at(index)) as ICellModel;
+  get(index: number): ICellModel {
+    return this._cellMap.get(this._cellOrder.get(index)) as ICellModel;
```
Comment (ellisonbg) [inline packages/notebook/src/celllist.ts]: Will we need to revisit this logic to make sure the two data structures are synchronized in this call sequence?

File: packages/notebook/src/celllist.ts [hunk 2]
```diff
@@ -231,33 +193,14 @@ class CellList implements IObservableUndoableVector<ICellModel> {
    * cell to the `CellList`. As such, `cell.dispose()` should
    * not be called by other actors.
    */
-  pushBack(cell: ICellModel): number {
+  push(cell: ICellModel): number {
```
Comment (ellisonbg) [inline packages/notebook/src/celllist.ts]: Same here, basically how do we know the cell map and list are syncrhonized.

File: packages/notebook/src/celllist.ts (hunk 1 above)
Comment (ian-r-rose) [inline packages/notebook/src/celllist.ts]: At least with the Google Drive implementation, this appears to be enough (the documentation is fairly light on that front). We may need to revisit it for other backends.

Comment (ellisonbg) [inline packages/notebook/src/celllist.ts]: OK great.

**Comment by ellisonbg** [comment]
Looks good, merging!

**Pull request merged**